src/java.desktop/share/classes/sun/font/FontManagerFactory.java

```diff
@@ -25,72 +25,33 @@
 
 package sun.font;
 
-import java.awt.AWTError;
-import java.awt.Font;
-import java.awt.GraphicsEnvironment;
-import java.awt.Toolkit;
-import java.security.AccessController;
-import java.security.PrivilegedAction;
-
-import sun.security.action.GetPropertyAction;
-
-
 /**
  * Factory class used to retrieve a valid FontManager instance for the current
  * platform.
  *
- * A default implementation is given for Linux, Solaris and Windows.
- * You can alter the behaviour of the {@link #getInstance()} method by setting
- * the {@code sun.font.fontmanager} property. For example:
- * {@code sun.font.fontmanager=sun.awt.X11FontManager}
+ * A default implementation is given for Linux, Mac OS and Windows.
  */
 public final class FontManagerFactory {
 
     /** Our singleton instance. */
-    private static FontManager instance = null;
-
-    private static final String DEFAULT_CLASS;
-    static {
-        if (FontUtilities.isWindows) {
-            DEFAULT_CLASS = "sun.awt.Win32FontManager";
-        } else if (FontUtilities.isMacOSX) {
-            DEFAULT_CLASS = "sun.font.CFontManager";
-            } else {
-            DEFAULT_CLASS = "sun.awt.X11FontManager";
-            }
-    }
+    private static volatile FontManager instance;
```

Comment:
Consider to instead of DCL to use the holder class pattern:

```
private static final class DeferredLoader {
  public static final FontManager INSTANCE = PlatformFontInfo.createFontManager();
}

public static FontManager getInstance() {
  return DeferredLoader.INSTANCE;
}
```